pages/learn/deploy/deployment.md

```diff
@@ -1,52 +1,250 @@
+projectName_service2                                             latest                 7bed253dada2        8 minutes ago       102MB
+```
 
-Then simply add your {{ $names.company.lower }} endpoint to your [git][git] repository via `git
-remote add {{ $names.company.short }} [application endpoint]`.
+__Note:__ by default docker image names need to be lower cases, so any `projectName` will be converted to lowercase as `projectname`.
+
+#### `--build, -b`                                
+This option on `balena deploy` will always force a build of the images before uploading and deploying. In the case when you don't specify the build option. `deploy` will use the images that already exist locally (you can see these by running `docker images`). Note that `--build` will not do a clean build every time and will make use of the local docker layer cache. If you want to do a full clean build you need to specify both the `--build` and `--nocache` flag (see below) 
+
+#### `--nocache`                                  
+The `--nocache` flag only applies when the `--build` flag is specified and it will cause Docker to build the images from scratch ignoring any layer cache from previous builds.
+
+#### `--buildArg, -B <arg>`                     
+Set a build-time variable (eg. "-B 'ARG=value'"). Can be specified multiple times. Warning: It is not recommended to use build-time variables for passing secrets like github keys, user credentials etc. Build-time variable values are visible to any user of the image with the docker history command. For this type of sensitive data it is recommended to use [build time secrets](#build-time-secrets-and-variables).
+
+#### `--emulated, -e`
+The `--emulated` flag enables you to run an emulated build using [qemu](https://www.qemu.org/) on your development machine. This should allow you to build `armv7l` binaries for devices like the Raspberry Pi on your development machine.
+
+#### `--logs`
+This option will stream the docker build log output for all your services to the terminal where you run deploy. These are the same logs that will be available on the [release logs page](#view-past-deployments). Note that if `deploy` is run without the `--build` flag, no logs will be output because no build will occur.
+
+#### `--nologupload`
+This option disables the uploading of all the service build logs to balenaCloud, so they will not be visible in [release logs page](#view-past-deployments). 
+
+#### `--source, -s <source>`
+The `--source` flag allows you do define a path to your source code folder. You should ensure your folder follows the [standard balena project structure](#project-structure).         
+
+## Git Push
+
+### Overview
+
+The `git push balena master` method of deployment was the original deployment mechanism for balenaCloud and while it will continue to always be supported and improved, the majority of our new feature work will be on improving [balena push](#balena-push).
```

Comment:
Suggest changing to:

The `git push balena master` method of deployment is the original deployment mechanism for balenaCloud. While we will continue to support and improve `git push`, our primary focus going forward will be to build upon [balena push](#balena-push).